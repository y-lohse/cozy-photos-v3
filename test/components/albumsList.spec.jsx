'use strict'

/* eslint-env jest */

import React from 'react'
import { shallow } from 'enzyme'

import { mockT, mockF } from 'cozy-ui/react/I18n'
import AlbumsList from '../../src/components/AlbumsList'

const mockAlbums = [
  {
    _type: 'io.cozy.photos.albums',
    name: 'albumTest2',
    _id: '33dda00f0eec15bc3b3c59a615001ac8'
  },
  {
    _type: 'io.cozy.photos.albums',
    name: 'albumTest2',
    _id: 'f717eb4d94f07737b168d3dbb7002141'
  }
]

describe('AlbumsList component', () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it('should be displayed correctly if albums', () => {
    const component = shallow(
      <AlbumsList t={mockT} f={mockF} albums={mockAlbums} />
    ).node
    expect(component).toMatchSnapshot()
  })

  it('should display empty component if albums list empty', () => {
    const component = shallow(
      <AlbumsList t={mockT} f={mockF} albums={[]} />
    ).node
    expect(component).toMatchSnapshot()
  })
})
