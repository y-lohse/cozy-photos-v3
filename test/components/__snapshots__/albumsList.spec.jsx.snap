exports[`AlbumsList component should be displayed correctly if albums 1`] = `
<DumbAlbumsList
  albums={
    Array [
      Object {
        "_id": "33dda00f0eec15bc3b3c59a615001ac8",
        "_type": "io.cozy.photos.albums",
        "name": "albumTest2",
      },
      Object {
        "_id": "f717eb4d94f07737b168d3dbb7002141",
        "_type": "io.cozy.photos.albums",
        "name": "albumTest2",
      },
    ]
  }
  f={[Function]}
  t={[Function]} />
`;

exports[`AlbumsList component should display empty component if albums list empty 1`] = `
<_translate
  emptyType="albums" />
`;
